app/models/date_range.rb

```diff
@@ -21,10 +21,10 @@ def compact
     
     same_years = @start_date.year == @end_date.year
     same_months = (@start_date.month == @end_date.month) && same_years
-    elements = elements.gsub(' %Y', '') if same_years
-    elements = elements.gsub(' %b', '') if same_months
-    
+    elements = elements.chomp(' %Y') if same_years
+    elements = elements.chomp(' %b') if same_months
```

Comment:
Theoretically, we can omit the `same year` check in `same months`, because `chomp` will only remove the month-substring if it is at the end of the string. 
I left it in though, because it is easier to understand. Agree?